src/pages/WalletManagement/WalletWelcomePage.tsx

```diff
@@ -16,27 +16,71 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
+import { deriveNewAddressData, TOTAL_NUMBER_OF_GROUPS } from 'alephium-js'
+import { Info } from 'lucide-react'
 import { useState } from 'react'
 import Confetti from 'react-confetti'
 import { useHistory } from 'react-router'
 import styled from 'styled-components'
 
 import Button from '../../components/Button'
+import ExpandableSection from '../../components/ExpandableSection'
+import InfoBox from '../../components/InfoBox'
+import KeyValueInput from '../../components/Inputs/InlineLabelValueInput'
+import Toggle from '../../components/Inputs/Toggle'
 import { FooterActionsContainer, Section } from '../../components/PageComponents/PageContainers'
 import Paragraph from '../../components/Paragraph'
+import { Address, useAddressesContext } from '../../contexts/addresses'
+import { useGlobalContext } from '../../contexts/global'
+import { getRandomLabelColor } from '../../utils/colors'
 import { useTimeout, useWindowSize } from '../../utils/hooks'
+import { openInWebBrowser } from '../../utils/misc'
 
 // This is shown when a user creates or imports a wallet
 
 const WalletWelcomePage = () => {
-  const { width, height } = useWindowSize()
+  const [shouldGenerateOneAddressPerGroup, setShouldGenerateOneAddressPerGroup] = useState(false)
   const [confettiRunning, setConfettiRunning] = useState(true)
+  const { wallet } = useGlobalContext()
+  const { width, height } = useWindowSize()
   const history = useHistory()
+  const { mainAddress, setAddress, saveNewAddress } = useAddressesContext()
 
   useTimeout(() => {
     setConfettiRunning(false)
   }, 3000)
 
+  const onButtonClick = () => {
+    if (shouldGenerateOneAddressPerGroup && wallet?.seed && mainAddress) {
+      const newAddresses = []
+      const color = getRandomLabelColor()
+
+      for (let group = 0; group < TOTAL_NUMBER_OF_GROUPS; group++) {
```

Comment:
We already have a `generateOneAddressPerGroup()` method in `NewAddressModal`. We could probably make it an export in `AddressesContextProvider` to go a bit DRYier?

Reply:
Good point, fixed now!